src/components/molecules/EventPaymentButton/EventPaymentButton.tsx

```diff
@@ -15,27 +19,51 @@ const EventPaymentButton: React.FunctionComponent<PropsType> = ({
   const [isLoading, setIsLoading] = useState(false);
   const [stripeError, setStripeError] = useState<string | undefined>();
 
+  useConnectUserPurchaseHistory();
```

Comment:
This and the has User already bought ticket the same as in the check payment is needed